web/packages/core/src/ruffle-player.ts

```diff
@@ -541,6 +536,39 @@ export class RufflePlayer extends HTMLElement {
         }
     }
 
+    private checkOptions(
+        options: string | URLLoadOptions | DataLoadOptions
+    ): URLLoadOptions | DataLoadOptions {
+        if (typeof options === "string") {
+            return { url: options };
+        }
+
+        const check: (
+            condition: boolean,
+            message: string
+        ) => asserts condition = (condition, message) => {
+            if (!condition) {
+                const error = new TypeError(message);
+                error.ruffleIndexError = PanicError.JavascriptConfiguration;
+                this.panic(error);
+                throw error;
+            }
+        };
+        check(
+            typeof options === "object",
```

Comment:
`typeof null` is "object", so we should also check for `null` (maybe `options instanceof Object` would be better here?).

Reply:
I added an explicit check for `options !== null`, like we had before. `options instanceof Object` is problematic since it might be unnecessarily too restrictive - objects created using `Object.create(null)` won't be accepted. Though that's a very esoteric case.